pkg/metadata/groupversion.go

```diff
@@ -87,8 +88,34 @@ func GVRtoGVK(gvr schema.GroupVersionResource) schema.GroupVersionKind {
 	}
 }
 
+func loadSpecialCases() map[string]string {
+	cases := map[string]string{}
+
+	// Example: GROUP_RESOURCE_MAP="groupalias=groupaliases,person=people"
+	envVal := os.Getenv("KRO_GROUP_RESOURCE_MAP")
+	if envVal == "" {
+		return cases
+	}
+
+	pairs := strings.Split(envVal, ",")
+	for _, p := range pairs {
+		kv := strings.SplitN(strings.TrimSpace(p), "=", 2)
+		if len(kv) == 2 {
+			cases[strings.ToLower(kv[0])] = strings.ToLower(kv[1])
+		}
+	}
+	return cases
+}
+
+var specialCases = loadSpecialCases()
+
 func GVKtoGVR(gvk schema.GroupVersionKind) schema.GroupVersionResource {
```

Comment:
I see what you're trying to do here, but wonder if it would make more sense for this to be a field on the RGD itself.

You could imagine something right [here](https://github.com/kubernetes-sigs/kro/blob/main/api/v1alpha1/resourcegraphdefinition_types.go#L58C2-L58C37), a new optional field like `Resource` or `PluralKind` or something. I'm not sure quite what the right name is.

Reply:
@matthchr @TheCodingSheikh how about leveraging simple schema syntax and doing something like:
```yaml
  kind: Peanut | plural=Peanuts
```